daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionCoder.scala

```diff
@@ -267,6 +279,9 @@ object TransactionCoder {
           .map(_.toImmArray)
 
         for {
+          rv <- if (!protoExe.hasReturnValue || (txVersion precedes minExerciseResult))
+            Right(None)
+          else decodeVal(protoExe.getReturnValue).map(Some(_))
```

Comment:
This will succeed if txVersion >= 6 and return_value, a required field, is absent.

Reply:
I pushed 08badbc2986b2e74a3acb7b5528ab0ae2268b036 to solve this.